### PR TITLE
fix EVP_PKEY

### PR DESCRIPTION
**Pull request by nogolang**
fix EVP_PKEY and old version still work

### CODE REVIEW

File: lib/resty/rsa.lua [hunk 1]
```diff
@@ -388,7 +389,13 @@ function _M.new(_, opts)
         end
     end
 
-    local size = C.EVP_PKEY_size(pkey)
+    --local size = C.EVP_PKEY_size(pkey)
+    local size
+    if not pcall(function () return C.EVP_PKEY_size(pkey) end) then
+        size = C.EVP_PKEY_get_size(pkey)
+    else
+        size=C.EVP_PKEY_size(pkey)
```
Comment (spacewander) [inline lib/resty/rsa.lua]:
```suggestion
        size = C.EVP_PKEY_size(pkey)
```

File: lib/resty/rsa.lua [hunk 2]
```diff
@@ -388,7 +389,13 @@ function _M.new(_, opts)
         end
     end
 
-    local size = C.EVP_PKEY_size(pkey)
+    --local size = C.EVP_PKEY_size(pkey)
+    local size
+    if not pcall(function () return C.EVP_PKEY_size(pkey) end) then
```
Comment (spacewander) [inline lib/resty/rsa.lua]: Let's use pcall like https://github.com/spacewander/lua-resty-rsa/blob/d4e22cfaa851eabbcc50029e20156ae236d733d8/lib/resty/rsa.lua#L151, so we only need to call once when loading the module.

File: lib/resty/rsa.lua [hunk 3]
```diff
@@ -388,7 +389,13 @@ function _M.new(_, opts)
         end
     end
 
-    local size = C.EVP_PKEY_size(pkey)
+    --local size = C.EVP_PKEY_size(pkey)
```
Comment (spacewander) [inline lib/resty/rsa.lua]: We can remove it now

File: lib/resty/rsa.lua [hunk 4]
```diff
@@ -388,7 +396,8 @@ function _M.new(_, opts)
         end
     end
 
-    local size = C.EVP_PKEY_size(pkey)
+   local size=evp_pkey_size(pkey)
```
Comment (spacewander) [inline lib/resty/rsa.lua]:
```suggestion
   local size = evp_pkey_size(pkey)
```

**Review by spacewander — APPROVED** [review]

**Comment by nogolang** [comment]
thank you, actually this is my first pull request

**Comment by spacewander** [comment]
> thank you, actually this is my first pull request

I hope you're having fun!

**Review by spacewander — APPROVED** [review]

**Comment by spacewander** [comment]
@nogolang 
Merged. Thanks!

**Pull request merged**